builtin/credential/aws-ec2/path_config_sts.go

```diff
@@ -0,0 +1,243 @@
+	}
+
+	return s.Put(entry)
+}
+
+// lockedSetAwsStsEntry creates or updates an STS role association with the given accountID
+// This method acquires the write lock before creating or updating the STS entry.
+func (b *backend) lockedSetAwsStsEntry(s logical.Storage, accountID string, stsEntry *awsStsEntry) error {
+	if accountID == "" {
+		return fmt.Errorf("missing AWS account ID")
+	}
+
+	if stsEntry == nil {
+		return fmt.Errorf("missing AWS STS Role ARN")
+	}
+
+	b.configMutex.Lock()
+	defer b.configMutex.Unlock()
+
+	return b.nonLockedSetAwsStsEntry(s, accountID, stsEntry)
+}
+
+// nonLockedAwsStsEntry returns the STS role associated with the given accountID.
+// This method does not acquire the read lock before returning information. If locking is
+// desired, use lockedAwsStsEntry instead
+func (b *backend) nonLockedAwsStsEntry(s logical.Storage, accountID string) (*awsStsEntry, error) {
+	entry, err := s.Get("config/sts/" + accountID)
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+	var stsEntry awsStsEntry
+	if err := entry.DecodeJSON(&stsEntry); err != nil {
+		return nil, err
+	}
+
+	return &stsEntry, nil
+}
+
+// lockedAwsStsEntry returns the STS role associated with the given accountID.
+// This method acquires the read lock before returning the association.
+func (b *backend) lockedAwsStsEntry(s logical.Storage, accountID string) (*awsStsEntry, error) {
+	b.configMutex.RLock()
+	defer b.configMutex.RUnlock()
+
+	return b.nonLockedAwsStsEntry(s, accountID)
+}
+
+// pathConfigStsRead is used to return information about an STS role/AWS accountID association
+func (b *backend) pathConfigStsRead(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	accountID := data.Get("account_id").(string)
+	if accountID == "" {
+		return logical.ErrorResponse("missing account id"), nil
+	}
+
+	stsEntry, err := b.lockedAwsStsEntry(req.Storage, accountID)
+	if err != nil {
+		return nil, err
+	}
+	if stsEntry == nil {
+		return nil, nil
+	}
+
+	return &logical.Response{
+		Data: structs.New(stsEntry).Map(),
+	}, nil
+}
+
+// pathConfigStsCreateUpdate is used to associate an STS role with a given AWS accountID
+func (b *backend) pathConfigStsCreateUpdate(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	accountID := data.Get("account_id").(string)
+	if accountID == "" {
+		return logical.ErrorResponse("missing AWS account ID"), nil
+	}
+
+	b.configMutex.Lock()
+	defer b.configMutex.Unlock()
+
+	// Check if an STS role is already registered
+	stsEntry, err := b.nonLockedAwsStsEntry(req.Storage, accountID)
+	if err != nil {
+		return nil, err
+	}
+	if stsEntry == nil {
+		stsEntry = &awsStsEntry{}
+	}
+
+	// Check that an STS role has actually been provided
+	stsRole, ok := data.GetOk("sts_role")
+	if ok {
+		stsEntry.StsRole = stsRole.(string)
+	} else if req.Operation == logical.CreateOperation {
+		stsEntry.StsRole = data.Get("sts_role").(string)
```

Comment:
I noticed in the docs that this is a required parameter. When the operation is create and when this value is not supplied, instead of trying to fetch the default value, we should error out here: `return logical.ErrorResponse("missing sts role"), nil`